app/models/library_entry.rb

```diff
@@ -207,16 +203,32 @@ def kind
   end
 
   after_create do
+    # Activity History Stat will be using this
+    library_event = LibraryEvent.create_for(:added, self)
     # Stat STI
     case kind
     when :anime
       Stat::AnimeGenreBreakdown.increment(user, self)
       Stat::AnimeAmountConsumed.increment(user, self)
       Stat::AnimeFavoriteYear.increment(user, self)
+      Stat::AnimeActivityHistory.increment(user, library_event)
     when :manga
       Stat::MangaGenreBreakdown.increment(user, self)
       Stat::MangaAmountConsumed.increment(user, self)
       Stat::MangaFavoriteYear.increment(user, self)
+      Stat::MangaActivityHistory.increment(user, library_event)
+    end
+  end
+
+  after_update do
+    # Activity History Stat will be using this
+    # TODO: refactor this somehow?
+    library_event = LibraryEvent.create_for(:updated, self)
+    case kind
+    when :anime
+      Stat::AnimeActivityHistory.increment(user, library_event)
```

Comment:
🤔 Trying to think _why_ I wouldn't do that. Will go over the code I did again and get back to you on this.